algorithms/src/snark/marlin/data_structures/proof.rs

```diff
@@ -233,35 +278,43 @@ pub struct Proof<E: PairingEngine> {
 impl<E: PairingEngine> Proof<E> {
     /// Construct a new proof.
     pub fn new(
-        batch_size: usize,
+        batch_sizes_map: BTreeMap<CircuitId, usize>,
+        total_instances: usize,
```

Comment:
We don't need `total_instances`, right? We can derive it as `let total_instances = batch_sizes_map.values().sum();`

Reply:
Removed